### DISCUSSION TITLE
UR Type Definition for BIP44 Accounts

### DISCUSSION THREAD

**Discussion by craigraw** [BlockchainCommons/Gordian-Developer-Community#24] [Specifications]
I've created a PR for a new UR data type: https://github.com/BlockchainCommons/Research/pull/50

The abstract and motivation is copied below. Any feedback appreciated!

### Abstract

This BCR describes a data format that promotes standards-based sharing of BIP44 account level xpubs and other information allowing devices to join wallets with little to no user interaction required.

### Motivation

A number of script types for both single and multisig have emerged, each with its own well known derivation path. Devices (especially hardware wallets) need to share the xpub at the correct derivation path for a particular script type in order to be successfully represented in most wallet software.

In the absence of a standard, the burden of selecting the script type on the device falls to the user, requiring technical knowledge and made more difficult by the often limited display and input capabilities. For multisig wallets, this choice must be repeated correctly on each device.

More commonly, a proprietary data format is used which bundles the information for several script types into a set, allowing the choice to be made once through the wallet software. The wallet software can then select the appropriate type from the set provided by the device. This BCR seeks to establish a standard format for creating this set using output descriptors to share BIP44 account level data at standard derivations.

**Comment by ChristopherA** [comment]
I'm willing to support this if we can get one other solid commitment from a wallet vendor want to support import/export of this CBOR in a UR or as part of a UR.

@craigraw can you share a bit more on how you plan to use for your wallet? What does the workflow look like?

**Reply by Fonta1n3** [reply]
> I'm willing to support this if we can get one other solid commitment from a wallet vendor want to support import/export of this CBOR in a UR or as part of a UR.
> 
> @craigraw can you share a bit more on how you plan to use for your wallet? What does the workflow look like?

I think what this achieves is a “wallet export/import” standard that all wallets can use. Currently its a major pain having to try and support each one individually.

**Reply by craigraw** [reply]
Yes @Fonta1n3  is correct.

Here is a simple workflow for the use case of adding a hardware wallet to wallet software (Specter, FN, Sparrow etc):

1. The user creates a wallet in the wallet software
2. The user chooses wallet details (singlesig, multisig, m/n, script type) or uses the defaults provided by the software.
3. The user selects an 'Export' function from the hardware wallet. No configuration is necessary, but there may optionally be an account number selection done.
4. The wallet software scans the QR provided by the hardware wallet and retrieves the appropriate xpub that matches the settings provided in step 1, along with the master fingerprint.

Steps 3 and 4 can be repeated for multisig setups with multiple hardware wallets.

**Comment by Fonta1n3** [comment]
Would be happy to support this in Fully Noded.

To be clear from the PR it seems this would support all major derivations not just bip44?

**Reply by craigraw** [reply]
Yes correct.

**Comment by stepansnigirev** [comment]
+1. Importing all the keys at once will be very convenient.
We will be happy to implement it in Specter Desktop and DIY.
